zephyr/lib/regions_mm.c

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Jakub Dabek <jakub.dabek@intel.com>
+ */
+
+#include <sof/lib/regions_mm.h>
+
+struct virtual_memory_heap
+	vm_heaps[CONFIG_MP_MAX_NUM_CPUS + VIRTUAL_REGION_COUNT];
+
+/**
+ * @brief Fill vm_heaps array with information from zephyr.
+ *
+ * Virtual memory regions calculated in zephyr are translated here
+ * to a struct that will keep all information on current allocations
+ * and virtual to physical mappings that are related to heaps.
+ * System heap is not a part of this information. It only refers to
+ * virtual first heaps.
+ * Has to be initialized after calculations for regions is done in zephyr.
+ */
+static int virtual_heaps_init(const struct device *unused)
+{
+	ARG_UNUSED(unused);
+
+	struct sys_mm_drv_region *virtual_memory_regions =
+		(struct sys_mm_drv_region *)sys_mm_drv_query_memory_regions();
+
+	for (size_t i = 0;
+		i < CONFIG_MP_MAX_NUM_CPUS + VIRTUAL_REGION_COUNT;
+		i++) {
+		vm_heaps[i].virtual_region = &virtual_memory_regions[i];
+
+		switch (virtual_memory_regions[i].attr)	{
```

Comment:
oh, so it's not just a bit set in the attribute, the attribute is actually equal to one of those bits. I thought you'd need to do something like
```
#define MEM_REG_ATTR_MASK (MEM_REG_ATTR_OPPORTUNISTIC_MEMORY | MEM_REG_ATTR_SHARED_HEAP | MEM_REG_ATTR_CORE_HEAP)
switch (virtual_memory_regions[i].attr & MEM_REG_ATTR_MASK) {
```
but if you don't - all the better